types/avocat/index.d.ts

```diff
@@ -0,0 +1,19 @@
+// Type definitions for avocat 2.0
+// Project: https://github.com/hapipal/avocat#readme
+// Definitions by: Tim Costa <https://github.com/timcosta>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import { ResponseObject } from '@hapi/hapi';
+import { Boom } from '@hapi/boom';
+
+export interface Options {
+    return?: boolean;
+    includeMessage?: boolean;
+}
+
+// the return type has to be any because the following scenarios exist:
+// 1. options.return = false - this will always throw or return false
+// 2. options.return = true - this will either return false or a Boom type
+// Due to scenario 2 and that behavior changes based on options, a return
+// type of any is the only way I see this working
+export function rethrow(err: ResponseObject | Boom, options?: Options): any;
```

Comment:
I initially thought we could use something like this: https://stackoverflow.com/questions/52817922/typescript-return-type-depending-on-parameter but since when `options.return` is `true` you can have the same type as `options.return` false or a `Boom` object, it doesn't work. You'd have the same problem as you mentioned in your comments.

Reply:
Yep, I tried all sorts of solutions before going with the `any` but even if i did `false | Boom` and hard casted to Boom on the receiving end it still was very unhappy, so we get the `any`. I'm glad the comments explained the reasoning well enough.